src/storage/txn/statistics.rs

```diff
@@ -0,0 +1,153 @@
+// Copyright 2016 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::time::{Instant, Duration};
+use std::collections::HashMap;
+use rand::{self, Rng};
+use std::u8;
+
+pub const ROLL_INTERVAL_SECS: u64 = 60 * 10;
+pub const CHECK_EXPIRED_INTERVAL_SECS: u64 = 60 * 60;
+pub const U8_MAX: u64 = u8::max_value() as u64;
+
+struct WriteStats {
+    pub history: u64,
+    current: u64,
+    pub last_active: Instant,
+}
+
+impl WriteStats {
+    pub fn new() -> WriteStats {
+        WriteStats {
+            history: 0,
+            current: 0,
+            last_active: Instant::now(),
+        }
+    }
+
+    pub fn roll(&mut self) {
+        self.history += self.current;
+        self.current = 0;
+    }
+
+    pub fn incr(&mut self, n: u64) {
+        self.current += n;
+    }
+
+    pub fn clear_history(&mut self) {
+        self.history = 0;
+    }
+}
+
+pub struct RegionsWriteStats {
+    // region_id -> write stats
+    stats: HashMap<u64, WriteStats>,
+}
+
+impl RegionsWriteStats {
+    pub fn new() -> RegionsWriteStats {
+        RegionsWriteStats { stats: HashMap::new() }
+    }
+
+    pub fn incr(&mut self, region_id: u64, n: u64) {
+        let stat = self.stats.entry(region_id).or_insert_with(WriteStats::new);
+        stat.incr(n);
+    }
+
+    pub fn roll_and_up(&mut self, base: u64) {
+        let mut rng = rand::thread_rng();
+        for (_, stat) in &mut self.stats {
+            stat.roll();
+
+            // We initialize the stat with random number between [0 - base]. If we just use base
+            // initialize the stat, A large number of regions that haven't been written yet might
+            // execute GC command at about the same time.
```

Comment:
It is unnecessary because it is already randomlized to decide whether GC command should run for each region, they won't execute at the same time.